src/v/cluster/topics_frontend.cc

```diff
@@ -195,10 +198,37 @@ ss::future<std::vector<topic_result>> topics_frontend::update_topic_properties(
       });
 }
 
+ss::future<std::error_code> topics_frontend::do_update_data_policy(
+  const topic_properties_update& update,
+  model::timeout_clock::time_point timeout) {
+    vassert(
+      update.properties.data_policy.value.has_value(),
+      "Data-policy property does not contain value");
+    switch (update.properties.data_policy.op) {
+    case incremental_update_operation::set:
+        co_return co_await _dp_frontend.local().create_data_policy(
+          update.tp_ns, update.properties.data_policy.value.value(), timeout);
+    case incremental_update_operation::remove:
+        co_return co_await _dp_frontend.local().clear_data_policy(
+          update.tp_ns, timeout);
+    case incremental_update_operation::none:
+        co_return std::error_code(cluster::errc::success);
+    }
+}
+
 ss::future<topic_result> topics_frontend::do_update_topic_properties(
   topic_properties_update update, model::timeout_clock::time_point timeout) {
     update_topic_properties_cmd cmd(update.tp_ns, update.properties);
     try {
+        if (update.properties.data_policy.value.has_value()) {
```

Comment:
is it possible to route data policy updates directly to the data policy frontend from the kafka layer rather than going through the topic front end?

Reply:
For me, it is simpler, because data_policy_frontend does not know anything about current leader and do not resend request to another node